packages/jest-jasmine2/src/index.js

```diff
@@ -95,7 +96,19 @@ function jasmine2(
   env.execute();
   return reporter
     .getResults()
-    .then(results => addSnapshotData(results, config, snapshotState));
+    .then(results => {
+      results.testResults.forEach(result => {
+        result.failureMessages = result.failureMessages.map(
+          failureMessage => {
+            const {message, stack} = separateMessageFromStack(failureMessage);
+            return message + formatStackTrace(
+              stack, config, results.testFilePath,
+            );
+          }
+        );
+      });
+      return addSnapshotData(results, config, snapshotState);
```

Comment:
addSnapshotData already iterates over all the results. Could we rename this function and merge this functionality?